### PR TITLE
Setup DNS according AY profile correctly when a static net configuration is provided to linuxrc

### PR DESCRIPTION
**Pull request by mchf**
[bsc#971175](https://bugzilla.suse.com/show_bug.cgi?id=971175), turning L3 patch into maintenance update

TODO: wait for maintenance request for SLE-12-SP1 confirmation

### CODE REVIEW

**Comment by teclator** [comment]
In general LGTM

File: src/clients/save_network.rb [hunk 1]
```diff
@@ -247,15 +247,27 @@ def copy_from_instsys
       nil
     end
 
+    # Creates DNS configuration on target
+    def configure_dns
+      if Mode.autoinst
+        NetworkAutoYast.instance.configure_dns
```
Comment (mvidner) [inline src/clients/save_network.rb]: Regarding the several sets of configuration data that YaST deals with, this method assigns the "AY profile" to the "YaST working set", but does not write it to the "persistent system config" or "current system config".
OTOH NetworkAutoconfiguration::configure_dns does write to the system.
It is inconsistent, and seems to contradict the comment above, "Creates DNS configuration on target".

File: src/lib/network/network_autoyast.rb [hunk 2]
```diff
@@ -120,7 +121,36 @@ def set_network_service
       NetworkService.EnableDisableNow
     end
 
-    private
+    # Initializates DNS setup according AY profile
+    #
+    # FIXME: it currently don't write DNS configuration. It is used for initialization
+    # of DNS setup according AY profile in 1st stage as part of network setup was moved
+    # here already and some parts of network configuration needs to know it. DNS write
+    # is still done in 2nd stage.
```
Comment (mvidner) [inline src/lib/network/network_autoyast.rb]: If a public method returns something, we should document what the return value means.

File: src/clients/save_network.rb [hunk 3]
```diff
@@ -247,15 +247,28 @@ def copy_from_instsys
       nil
     end
 
+    # Creates target's default DNS configuration
+    #
+    # It uses DNS configuration as defined in AY profile (if any) or
+    # proposes a predefined default values.
+    def configure_dns
+      ret = false
```
Comment (mvidner) [inline src/clients/save_network.rb]: "`ret`" means "the value to be returned from this method", so the name is wrong here. `configured` instead?

**Comment by mvidner** [comment]
Thank you!
LGTM

**Comment by mchf** [comment]
Thanks for reviews. I'll merge it once corresponding PTF is accepted.

**Pull request merged**